File: euler_7.js
//old relic file, for posterity


var primes = []
var number = {
	value : 1,
	prime : true
}

while (primes.length < 10001) {
	number.value ++;
	number.prime = true;
	number.sqrt = Math.sqrt(number.value);
	for (var i = 2; i <= number.sqrt; i++){
		if (number.value % i == 0) {
			number.prime = false;
		}
	}
	if (number.prime == true) {
		primes.push(number.value);
	}
}

console.log(primes[primes.length - 1]);

File: euler_7.py
import math
def main():
    n = 1
    primes_count = 0
    while n:
        n += 1
        for a in range(2, int(math.floor(math.sqrt(n))) + 1):
            if n % a == 0:
                break
        else:
            primes_count += 1
            print primes_count

        if primes_count == 10001:
            break

    return n

if __name__ == '__main__':
    main()
